server/src/internalClusterTest/java/org/elasticsearch/search/fieldcaps/FieldCapabilitiesIT.java

```diff
@@ -913,6 +916,154 @@ public void testIndexMode() throws Exception {
         assertThat(actualIndexModes, equalTo(indexModes));
     }
 
+    public void testResolvedExpressionWithIndexAlias() {
+        FieldCapabilitiesResponse response = client().prepareFieldCaps("current").setFields("*").setIncludeResolvedTo(true).get();
+        assertIndices(response, "new_index");
+
+        assertEquals(0, response.getResolvedRemotely().size());
+        ResolvedIndexExpressions resolvedLocally = response.getResolvedLocally();
+        List<ResolvedIndexExpression> expressions = resolvedLocally.expressions();
+        assertEquals(1, resolvedLocally.expressions().size());
+        ResolvedIndexExpression expression = expressions.get(0);
+        assertEquals("current", expression.original());
+        Set<String> concreteIndices = expression.localExpressions().expressions();
+        assertEquals(1, concreteIndices.size());
+        assertTrue(concreteIndices.contains("new_index"));
+    }
+
+    public void testResolvedExpressionWithWildcard() {
+        FieldCapabilitiesResponse response = client().prepareFieldCaps("*index").setFields("*").setIncludeResolvedTo(true).get();
+        assertIndices(response, "new_index", "old_index");
+
+        assertEquals(0, response.getResolvedRemotely().size());
+        ResolvedIndexExpressions resolvedLocally = response.getResolvedLocally();
+        List<ResolvedIndexExpression> expressions = resolvedLocally.expressions();
+        assertEquals(1, resolvedLocally.expressions().size());
+        ResolvedIndexExpression expression = expressions.get(0);
+        assertEquals("*index", expression.original());
+        Set<String> concreteIndices = expression.localExpressions().expressions();
```

Comment:
This was introduced in another pr, but I find naming a bit confusing.

`response.getResolvedLocally().expressions()` -> I assume those are expressions originally supplied by the user, all good here
`response.getResolvedLocally().expressions().getFirst().localExpressions().expressions()` -> the last 2 expressions are a bit confusing. Should it be something like `response.getResolvedLocally().expressions().getFirst().resolution().concreteIndices()`? (and corresponding `LocalExpressions` renamed to `LocalIndexResolution`). Maybe I misunderstand the content/meaning of the data structure.